src/utils/quick-open.js

```diff
@@ -49,6 +50,56 @@ export function parseLineColumn(query: string) {
   }
 }
 
+export function formatSourcesForProjectDirectoryRoot(
+  source: Source,
+  projectDirectoryRoot: string
+) {
+  const sourcePath = getSourcePath(source.get("url"));
+  let title = "";
+  let subtitle = "";
+
+  if (sourcePath) {
+    const sourcePathSplit = sourcePath.split("/");
+
+    // Remove leading "/"
+    if (sourcePathSplit[0] === "") {
+      sourcePathSplit.shift();
+    }
+
+    // Make the title the file name
+    title = sourcePathSplit.pop().split("?")[0];
+
+    // Rebuild the sourcePath
+    let newSourcePath = sourcePathSplit.join("/");
+
+    if (projectDirectoryRoot != "") {
```

Comment:
This should be in a helper too

```js
truncateFilePathRoot(path, root)
```

this ofcourse could be tested in several different ways too